libsolidity/parsing/Parser.cpp

```diff
@@ -626,10 +630,11 @@ ASTPointer<Mapping> Parser::parseMapping()
 	ASTNodeFactory nodeFactory(*this);
 	expectToken(Token::Mapping);
 	expectToken(Token::LParen);
-	if (!Token::isElementaryTypeName(m_scanner->currentToken()))
-		fatalParserError(string("Expected elementary type name for mapping key type"));
 	ASTPointer<ElementaryTypeName> keyType;
-	keyType = ASTNodeFactory(*this).createNode<ElementaryTypeName>(m_scanner->currentToken());
+	if (ElementaryTypeNameToken::isElementaryTypeName(m_scanner->currentLiteral()))
```

Comment:
potentially, yes...I don't see what the purpose would be of that though, is it really so imperative that the scanner (which can already detect whether or not it is in the token list by means of checking the literal) be modified to suite the needs of prechecking that it can be a token?